dipy/align/imaffine.py

```diff
@@ -921,16 +933,22 @@ def _init_optimizer(self, static, moving, transform, params0,
         n = transform.get_number_of_parameters()
         self.nparams = n
 
+        static_masked, moving_masked = static, moving
```

Comment:
by doing so `static_masked` and `static` will be both same? and the same goes for `moving` and `moving_masked`

Reply:
okay, I see, if the mask is empty we use original data and if the mask is available we mask the data and use that for registration.

Reply:
So the registration parameters will be optimized for masked data (say mask is provided) and if someone applies transformation to original data (not masked) will it give good output?

I guess we need to check it by testing it.

Reply:
What do you mean by good output?
As it is affine registration, the transformation of out-of-mask data should not go completely crazy as it could with warping field. Of course registering masked brain would not register well other non-rigidly attached body-parts such as neck and spine.

Reply:
I am saying we should test different senarios and compare the results.